orm/models.py

```diff
@@ -216,11 +216,16 @@ def limit(self, limit_count: int):
     async def create(self, **kwargs):
         # Validate the keyword arguments.
         fields = self.model_cls.fields
+        pkname = self.model_cls.__pkname__
+        pk = fields[pkname]
         required = [key for key, value in fields.items() if not value.has_default()]
         validator = typesystem.Object(
             properties=fields, required=required, additional_properties=False
         )
         kwargs = validator.validate(kwargs)
+        # remove pk with NULL value to avoid constraint violation in postgres
+        if pk.allow_null:
+            del kwargs[pkname]
```

Comment:
@kiddick 
I think this will ignore the pk value provided by client.

for example:

    class User(orm.Model):
        __tablename__ = "users"

        id = orm.Integer(primary_key=True)
        name = orm.String(max_length=100)


    user = await User.objects.create(id=100000001, name="Tom")
    assert user.id == 100000001

Reply:
I add a test for this case in kiddick/orm#1